src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/InitializationService.java

```diff
@@ -0,0 +1,315 @@
+    private final ClientService clientService;
+    private final SignerProxyFacade signerProxyFacade;
+
+    @Setter
+    private boolean isTokenPinEnforced = SystemProperties.shouldEnforceTokenPinPolicy();
+
+    @Autowired
+    public InitializationService(SystemService systemService, ServerConfService serverConfService,
+            TokenService tokenService, GlobalConfFacade globalConfFacade, ClientService clientService,
+            SignerProxyFacade signerProxyFacade) {
+        this.systemService = systemService;
+        this.serverConfService = serverConfService;
+        this.tokenService = tokenService;
+        this.globalConfFacade = globalConfFacade;
+        this.clientService = clientService;
+        this.signerProxyFacade = signerProxyFacade;
+    }
+
+    /**
+     * Check the whole init status of the Security Server. The init status consists of the following:
+     * 1. is anchor imported - whether or not a configuration anchor has been imported
+     * 2. is server conf initialized - whether or not a server conf exists
+     * 3. is software token initialized - whether or not a software token exists AND
+     * it's status != TokenStatusInfo.NOT_INITIALIZED
+     * @return
+     */
+    public InitializationStatusDto isSecurityServerInitialized() {
+        boolean isAnchorImported = systemService.isAnchorImported();
+        boolean isServerConfInitialized = serverConfService.isServerConfInitialized();
+        boolean isSoftwareTokenInitialized = tokenService.isSoftwareTokenInitialized();
+        InitializationStatusDto initializationStatusDto = new InitializationStatusDto();
+        initializationStatusDto.setAnchorImported(isAnchorImported);
+        initializationStatusDto.setServerConfInitialized(isServerConfInitialized && isSoftwareTokenInitialized);
+        return initializationStatusDto;
+    }
+
+    /**
+     * Initialize a new Security Server with the provided parameters. The method will throw an exception if the
+     * server has already been initialized. The method does support partial initialization (e.g. server conf has
+     * been created but software token has not) BUT taking that into use will require changing
+     * {@link #verifyInitializationPrerequisites()} to throw UnhandledWarningsException with warning metadata
+     * instead of InitializationException.
+     * @param securityServerCode server code for the new Security Server
+     * @param ownerMemberClass member class of the new owner member
+     * @param ownerMemberCode member code of the new owner member
+     * @param softwareTokenPin pin code for the initial software token (softToken-0)
+     * @param ignoreWarnings
+     * @throws AnchorNotFoundException if an anchor has not been imported
+     * @throws InitializationException if prerequisite check fails: if a server conf already exists OR if a
+     * software token has already been initialized
+     * @throws UnhandledWarningsException
```

Comment:
Maybe a couple of words about ignoreWarnings and UnhandledWarningsException would be good, since it is a kind of complicated concept.

Reply:
Added some better explanations in javadocs